Add message content access notice for affected fields (#4045)
hammerandchisel/discord-api-docs: docs/resources/Channel.md

| mentions\*\*\*                | array of [user](#DOCS_RESOURCES_USER/user-object) objects, with an additional partial [member](#DOCS_RESOURCES_GUILD/guild-member-object) field | users specifically mentioned in the message                                                                                                     |
| mention_roles                 | array of [role](#DOCS_TOPICS_PERMISSIONS/role-object) object ids                                                                                | roles specifically mentioned in this message                                                                                                    |
| mention_channels?\*\*\*\*     | array of [channel mention](#DOCS_RESOURCES_CHANNEL/channel-mention-object) objects                                                              | channels specifically mentioned in this message                                                                                                 |

> | attachments\*\*\*\*\*\*             | array of [attachment](#DOCS_RESOURCES_CHANNEL/attachment-object) objects                                                                        | any attached files                                                                                                                              |
> | embeds\*\*\*\*\*\*                  | array of [embed](#DOCS_RESOURCES_CHANNEL/embed-object) objects                                                                                  | any embedded content                                                                                                                            |

Reviewer: ```suggestion
| attachments\*\*\*\*\*\*       | array of [attachment](#DOCS_RESOURCES_CHANNEL/attachment-object) objects                                                                        | any attached files                                                                                                                              |
| embeds\*\*\*\*\*\*            | array of [embed](#DOCS_RESOURCES_CHANNEL/embed-object) objects                                                                                  | any embedded content                                                                                                                            |
```

| referenced_message?\*\*\*\*\* | ?[message object](#DOCS_RESOURCES_CHANNEL/message-object)                                                                                       | the message associated with the message_reference                                                                                               |
| interaction?                  | [message interaction object](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/message-interaction-object-message-interaction-structure)              | sent if the message is a response to an [Interaction](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING/)                                             |
| thread?                       | [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object                                                                                        | the thread that was started from this message, includes [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) object                    |

> | components?\*\*\*\*\*\*             | Array of [message components](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/component-object)                                                           | sent if the message contains components like buttons, action rows, or other interactive components                                              |

Reviewer: ```suggestion
| components?\*\*\*\*\*\*       | array of [message components](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/component-object)                                                           | sent if the message contains components like buttons, action rows, or other interactive components                                              |
```

\*\*\*\*\* This field is only returned for messages with a `type` of `19` (REPLY) or `21` (THREAD_STARTER_MESSAGE). If the message is a reply but the `referenced_message` field is not present, the backend did not attempt to fetch the message that was being replied to, so its state is unknown. If the field exists but is null, the referenced message was deleted.

> \*\*\*\*\*\* If your bot or app is not approved for message content these fields will either be an empty string or empty array, depending on the data type. **These restrictions will not apply to message objects received via interaction payloads** (Only for verified bots.)

Reviewer 2: ```suggestion
\*\*\*\*\*\* If the Message Content Intent is not enabled for your bot, these fields will either be an empty string or empty array, depending on the data type. These fields are still always accessible for messages the bot sends, messages the bot receives in DMs or messages in which the bot is mentioned. [Learn more here](https://support-dev.discord.com/hc/en-us/articles/4404772028055).
```
